splice_machine/src/main/java/com/splicemachine/pipeline/foreignkey/ForeignKeyParentInterceptWriteHandler.java

```diff
@@ -42,35 +65,261 @@
+            readUncommittedFilter = new SimpleTxnFilter(Long.toString(indexConglomerateId), ((ActiveWriteTxn) txnView).getReadUncommittedActiveTxn(), NoOpReadResolver.INSTANCE, SIDriver.driver().getTxnStore());
+
+        } else if (txnView instanceof WritableTxn) {
+            readCommittedFilter = new SimpleTxnFilter(Long.toString(indexConglomerateId), ((WritableTxn) txnView).getReadCommittedActiveTxn(), NoOpReadResolver.INSTANCE, SIDriver.driver().getTxnStore());
+            readUncommittedFilter = new SimpleTxnFilter(Long.toString(indexConglomerateId), ((WritableTxn) txnView).getReadUncommittedActiveTxn(), NoOpReadResolver.INSTANCE, SIDriver.driver().getTxnStore());
+        } else {
+            throw new IOException("invalidTxn,");
         }
-        ctx.sendUpstream(mutation);
+        return Pair.newPair(readCommittedFilter, readUncommittedFilter);
     }
+
+    private byte[] isVisible(List<DataCell> next, SimpleTxnFilter txnFilter) throws IOException {
+        int cellCount = next.size();
+        for(DataCell dc:next){
+            DataFilter.ReturnCode rC = txnFilter.filterCell(dc);
+            switch(rC){
+                case NEXT_ROW:
+                    return null; //the entire row is filtered
+                case SKIP:
+                case NEXT_COL:
+                case SEEK:
+                    cellCount--; //the cell is filtered
+                    break;
+                case INCLUDE:
+                case INCLUDE_AND_NEXT_COL: //the cell is included
+                default:
+                    break;
+            }
+        }
+        if(cellCount > 0) {
+            return next.get(0).key();
+        }
+        return null;
+    }
+
+    private static byte[] toChildBaseRowId(byte[] indexRowId, DDLMessage.FKConstraintInfo fkConstraintInfo) {
+        // 1. determine the position
+        MultiFieldDecoder multiFieldDecoder = MultiFieldDecoder.create();
+        TypeProvider typeProvider = VersionedSerializers.typesForVersion(fkConstraintInfo.getParentTableVersion());
+        int position = 0;
+        multiFieldDecoder.set(indexRowId);
+        for (int i = 0; i < fkConstraintInfo.getFormatIdsCount(); i++) {
+            if (multiFieldDecoder.nextIsNull()) {
+                return null;
+            }
+            if (fkConstraintInfo.getFormatIds(i) == StoredFormatIds.SQL_DOUBLE_ID) {
+                position += multiFieldDecoder.skipDouble();
+            } else if (fkConstraintInfo.getFormatIds(i) == StoredFormatIds.SQL_REAL_ID) {
+                position += multiFieldDecoder.skipFloat();
+            } else if (typeProvider.isScalar(fkConstraintInfo.getFormatIds(i))) {
+                position += multiFieldDecoder.skipLong();
+            } else {
+                position += multiFieldDecoder.skip();
+            }
+        }
+        int lastKeyIndex = position - 2;
+
+        if (lastKeyIndex == indexRowId.length - 1) {
+            return null; // special case.
```

Comment:
Maybe a bit more description about the special case?

Reply:
This is an outdated version of this method. Check out the new version of it in `OnDeleteAbstractAction`, we throw if we encounter this special code path.